ding/entry/serial_entry_offline.py

```diff
@@ -72,17 +88,18 @@ def serial_pipeline_offline(
     learner.call_hook('before_run')
     stop = False
 
-    for epoch in range(cfg.policy.learn.train_epoch):
-        # Evaluate policy performance
-        for i, train_data in enumerate(dataloader):
-            if evaluator.should_eval(learner.train_iter):
-                stop, reward = evaluator.eval(learner.save_checkpoint, learner.train_iter)
-                if stop:
-                    break
+    for epoch in tqdm(range(cfg.policy.learn.train_epoch)):
+        if cfg.policy.learn.multi_gpu:
+            dataloader.sampler.set_epoch(epoch)
+        for train_data in tqdm(dataloader):
             learner.train(train_data)
-            if learner.train_iter >= max_train_iter:
-                stop = True
-        if stop:
+
+        # Evaluate policy at most once per epoch.
+        if get_rank() == 0 and evaluator.should_eval(learner.train_iter):
+            stop, reward = evaluator.eval(learner.save_checkpoint, learner.train_iter)
```

Comment:
there maybe a bug for multi-gpu, only rank0 can get `stop` information, you should broadcast this value to other rank

Reply:
changed